scripts/install_binder.sh

```diff
@@ -1,31 +1,54 @@
 #!/bin/bash
 set -e
 
-## force install of last working version of rstudio, if necessary
-# RSTUDIO_VERSION=1.3.959 /rocker_scripts/install_rstudio.sh
+## build ARGs
+NCPUS=${NCPUS:-"-1"}
+
+NB_USER=${NB_USER:-${DEFAULT_USER:-"rstudio"}}
+
+# a function to install apt packages only if they are not installed
+function apt_install() {
+    if ! dpkg -s "$@" >/dev/null 2>&1; then
+        if [ "$(find /var/lib/apt/lists/* | wc -l)" = "0" ]; then
+            apt-get update
+        fi
+        apt-get install -y --no-install-recommends "$@"
+    fi
+}
+
+apt_install git sudo libzmq3-dev
+
+# set up the default user if it does not exist
+if ! id -u "${NB_USER}" >/dev/null 2>&1; then
+    /rocker_scripts/default_user.sh "${NB_USER}"
+fi
+
+# install python
+/rocker_scripts/install_python.sh
 
-## NOTE: this runs as user NB_USER!
 PYTHON_VENV_PATH=${PYTHON_VENV_PATH:-/opt/venv/reticulate}
-DEFAULT_USER=${DEFAULT_USER:-rstudio}
-NB_USER=${NB_USER:-${DEFAULT_USER}}
-NB_UID=${NB_UID:-1000}
 WORKDIR=${WORKDIR:-/home/${NB_USER}}
-usermod -l "${NB_USER}" "${DEFAULT_USER}"
 # Create a venv dir owned by unprivileged user & set up notebook in it
 # This allows non-root to install python libraries if required
 mkdir -p "${PYTHON_VENV_PATH}"
 chown -R "${NB_USER}" "${PYTHON_VENV_PATH}"
 
+# to use pyenv in a RStudio session, we need to include the PATH in the .profile file
+# https://github.com/rocker-org/rocker-versioned2/issues/428
 PATH=/opt/pyenv/bin:${PATH}
-
-# And set ENV for R! It doesn't read from the environment...
-echo "PATH=${PATH}" >>"${R_HOME}/etc/Renviron.site"
 echo "export PATH=${PATH}" >>"${WORKDIR}/.profile"
 
 cd "${WORKDIR}"
 ## This gets run as user
 sudo -u "${NB_USER}" python3 -m venv "${PYTHON_VENV_PATH}"
```

Comment:
I do not understand the need for this line.
My understanding is that this environment is not used within this script...

Reply:
It seems that `install_python.sh` also has unused code related to `venv` or `virtualenv`.
Do you think it is OK to delete these all together as well?

Reply:
probably. do you want to tackle updates to install_python.sh here?

Reply:
This PR set up tests for each script and it make it easier future changes, so I would like to remove the virtual envs at another time.
I would like to merge this PR into #441 first to make sure it passes all tests and then into the default branch.